### PR TITLE
#3481: Added missing indexes for queries on tables related to saving …

### PR DESCRIPTION
**Pull request by ssferrazza**
Added missing indexes for queries on tables related to saving and reading a case.

### CODE REVIEW

**Comment by DavidBaldsiefen** [comment]
Can you also add an index to `users.uuid` and `user.userName`? Also, the two indexes I asked you to add, maybe it's better if we don't define `USING HASH` for them.

**Comment by ssferrazza** [comment]
@DavidBaldsiefen I'll add the two indexes to `users.uuid` and `user.userName`.

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 1]
```diff
@@ -6496,4 +6496,17 @@ ALTER TABLE cases_history
 
 INSERT INTO schema_version (version_number, comment) VALUES (326, 'SurvNet Adaptations - Create new field “Reinfection” for cases #3831');
 
+CREATE INDEX IF NOT EXISTS idx_samples_associatedcase_id ON samples USING btree (associatedcase_id);
+CREATE INDEX IF NOT EXISTS idx_eventparticipant_reporting_user_id ON eventparticipant USING btree (reportinguser_id);
+CREATE INDEX IF NOT EXISTS idx_cases_reporting_user_id ON cases USING hash (reportinguser_id);
+CREATE INDEX IF NOT EXISTS idx_cases_person_id ON cases USING btree (person_id);
+CREATE INDEX IF NOT EXISTS idx_contacts_reporting_user_id ON contact USING btree (reportinguser_id);
```
Comment (DavidBaldsiefen) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:6503]: Change name to `idx_contact_reporting_user_id`

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 2]
```diff
@@ -6496,4 +6496,17 @@ ALTER TABLE cases_history
...
+CREATE INDEX IF NOT EXISTS idx_diseaseconfig_changedate on diseaseconfiguration (changedate DESC);
```
Comment (DavidBaldsiefen) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:6504]: Change name to `idx_diseaseconfiguration_changedate`

**Review by DavidBaldsiefen — CHANGES_REQUESTED** [review]

**Comment by DavidBaldsiefen** [comment]
After that you will need to merge sormas_schema.sql, so you will propably need to increase the schema_version number

File: sormas-backend/src/main/resources/sql/sormas_schema.sql [hunk 3]
```diff
@@ -6508,4 +6508,18 @@ ALTER TABLE exposures_history ALTER COLUMN location_id DROP NOT NULL;
 UPDATE exposures SET location_id = null FROM cases WHERE cases.epidata_id = exposures.epidata_id AND cases.deleted IS true AND (SELECT COUNT(location_id) FROM exposures ex WHERE ex.location_id = exposures.location_id) > 1;
 
 INSERT INTO schema_version (version_number, comment) VALUES (328, 'Remove locations assigned to more than one exposure from deleted cases #4338');
+
```
Comment (DavidBaldsiefen) [inline sormas-backend/src/main/resources/sql/sormas_schema.sql:6511]: Insert comment including date (see code above for examples)

**Review by DavidBaldsiefen — CHANGES_REQUESTED** [review]

**Review by DavidBaldsiefen — APPROVED** [review]

**Pull request merged**